activity/kafkapub/activity.go

```diff
@@ -0,0 +1,171 @@
+	} else {
+		return fmt.Errorf("Kafkapub activity is not configured with at least one BrokerUrl")
+	}
+	if context.GetInput("Topic") != nil {
+		a.topic = context.GetInput("Topic").(string)
+		flogoLogger.Debugf("Kafkapub topic [%s]", a.topic)
+	} else {
+		return fmt.Errorf("Topic input parameter not provided and is required")
+	}
+
+	//clientKeystore
+	/*
+		Its worth mentioning here that when the keystore for kafka is created it must support RSA keys via
+		the -keyalg RSA option.  If not then there will be ZERO overlap in supported cipher suites with java.
+		see:   https://issues.apache.org/jira/browse/KAFKA-3647
+		for more info
+	*/
+	if context.GetInput("truststore") != nil {
+		trustStore := context.GetInput("truststore")
+		if trustStore != nil && len(trustStore.(string)) > 0 {
+			trustPool, err := getCerts(trustStore.(string))
+			if err != nil {
+				return err
+			}
+			config := tls.Config{
+				RootCAs:            trustPool,
+				InsecureSkipVerify: true}
+			a.kafkaConfig.Net.TLS.Enable = true
+			a.kafkaConfig.Net.TLS.Config = &config
+		}
+		flogoLogger.Debugf("Kafkapub initialized truststore from [%s]", trustStore)
+	}
+	// SASL
+	if context.GetInput("user") != nil {
+		var password string
+		user := context.GetInput("user").(string)
+		if context.GetInput("password") != nil {
+			password = context.GetInput("password").(string)
+		}
+		a.kafkaConfig.Net.SASL.Enable = true
+		a.kafkaConfig.Net.SASL.User = user
+		a.kafkaConfig.Net.SASL.Password = password
+		flogoLogger.Debugf("Kafkapub SASL parms initialized; user [%s]  password[########]", user)
+	}
+
+	syncProducer, err := sarama.NewSyncProducer(a.brokers, a.kafkaConfig)
+	if err != nil {
+		return fmt.Errorf("Kafkapub failed to create a SyncProducer.  Check any TLS or SASL parameters carefully.  Reason given: [%s]", err)
+	}
+	a.syncProducer = syncProducer
+
+	flogoLogger.Debug("Kafkapub synchronous producer created")
+	return nil
+}
+
+func getCerts(trustStore string) (*x509.CertPool, error) {
+	certPool := x509.NewCertPool()
+	fileInfo, err := os.Stat(trustStore)
+	if err != nil {
+		return certPool, fmt.Errorf("Truststore [%s] does not exist", trustStore)
+	}
+	switch mode := fileInfo.Mode(); {
+	case mode.IsDir():
+		break
+	case mode.IsRegular():
+		return certPool, fmt.Errorf("Truststore [%s] is not a directory.  Must be a directory containing trusted certificates in PEM format",
+			trustStore)
+	}
+	trustedCertFiles, err := ioutil.ReadDir(trustStore)
+	if err != nil || len(trustedCertFiles) == 0 {
+		return certPool, fmt.Errorf("Failed to read trusted certificates from [%s]  Must be a directory containing trusted certificates in PEM format", trustStore)
+	}
+	for _, trustCertFile := range trustedCertFiles {
+		fqfName := fmt.Sprintf("%s%c%s", trustStore, os.PathSeparator, trustCertFile.Name())
+		trustCertBytes, err := ioutil.ReadFile(fqfName)
+		if err != nil {
+			flogoLogger.Warnf("Failed to read trusted certificate [%s] ... continuing", trustCertFile.Name())
```

Comment:
Do you want to call continue here? ioutil.ReadFile can return nil and you will be passing that to AppendCertsFromPEM, no real problem, just some extra computation